### PR TITLE
Add block editor support for profiles

### PR DESCRIPTION
**Pull request by joemcgill**
This changes the editor used for profile posts to the block editor. This is done by setting `show_in_rest` to `true` when registering the post type. Since this post type is already public and set to use `post` caps, it should not expose any data that isn't already available through other means.

### CODE REVIEW

**Review by mboynes — APPROVED** [review]
💯

**Comment by joemcgill** [comment]
Looks like the post meta field for searching/selecting an attached author isn't working in the block editor currently, so that will need to be addressed before this is merged. Ideally, I think we should move the control into a sidebar panel instead of the generic post meta fields panel, but iteration would be ok if needed.

**Comment by emilyatmobtown** [comment]
Fixed the byline selector in #76 and merged into this PR.

File: inc/rest-api.php [hunk 1]
```diff
@@ -165,6 +165,24 @@ function rest_user_search( \WP_REST_Request $request ) {
...
+ * @param \WP_REST_Request $request Request used to generate the response.
+ * @return mixed Filtered response.
+ */
+function filter_rest_pre_dispatch( $result, $server, $request ) {
+	if ( '/byline-manager/v1/authors' === $request->get_route() ) {
+		add_filter( 'ep_skip_query_integration', '__return_true' );
+	}
+	return $result;
+}
+add_filter( 'rest_pre_dispatch', __NAMESPACE__ . '\filter_rest_pre_dispatch', 10, 3 );
```
Comment (renatonascalves) [inline inc/rest-api.php]: I have contrary feedback about this change: as an open-source plugin, I don't think we should be making this decision for all users.

If I decide to use the VIP Enterprise Search here, this is creating a new task for me to unhook this. I could be unaware of this change, as most users will be, and assume it is using Elasticsearch when it is not.

The root of the issue stands for this line, https://github.com/alleyinteractive/byline-manager/blob/a51e6d27b38cdc3a3d2bd5a879303077317cdcf6/inc/rest-api.php#L64, together with the `	define( 'VIP_ENABLE_VIP_SEARCH_QUERY_INTEGRATION', true );` flag VIP provides, which forces all search queries to use VIP Search.

If I set this to `false`, I think it should stop this from happening. Essentially, someone is making a decision to run ES in all search queries unaware of its reach.

--

My suggestion would be to not add this to the plugin and leave the site itself to handle the ElasticSearch integration (either removing it as you did or indexing the BM post type).

**Review by renatonascalves — APPROVED** [review]
My two cents! 🕺🏾

File: inc/rest-api.php (hunk 1 above)
Comment (emilyatmobtown) [reply inc/rest-api.php]: @renatonascalves @joemcgill - Personally, I don't have a dog in this fight. I can see value in either approach. However, I would like to get the REST API support change merged to `production` to complete Block Editor support. This is currently a blocker on another project. Recognizing that the broken Byline Selector that Joe noted earlier in this PR is due to ES integration, I'd suggest we isolate that as a separate conversation to be continued apart from this PR. I'm pulling that change out of this PR.

**Comment by renatonascalves** [comment]
I'd like to suggest two new changes that could be bundled as part of this pull request.

- Make sure the packages can be installed when using PHP 8: https://github.com/alleyinteractive/byline-manager/blob/production/composer.json#L13
- Remove the `composer.lock` file

**Comment by joemcgill** [comment]
Thanks for your feedback, @renatonascalves. Let me review the approach and see if there's a less opinionated way that we could avoid the broken behavior in the editor.

In terms of your other suggestions, I agree that we need to update the `composer.json` file to be PHP 8 compatible, but that seems like an unrelated change that can be handled in a separate PR. Since the only dependencies of this project are dev-dependencies, removing the `composer.lock` file is probably ok as well, but could cause unrelated changes to automated tests if an underlying PHPCS or PHPUnit dependency changes underneath us, so I'm a bit hesitant to do so without cause.

**Comment by joemcgill** [comment]
I've tested the latest updates on a default install and [the original problem I noted](https://github.com/alleyinteractive/byline-manager/pull/70#issuecomment-1087829121) with the search box not working is resolved, so I'm going to go ahead and merge this and we can follow up on the ES integration issues and other dependency updates in follow-up tasks.

**Pull request merged**